hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/replication/TestReplicationSupervisor.java

```diff
@@ -173,6 +178,36 @@ public void stalledDownload() {
     }
   }
 
+  @Test
+  public void testDownloadAndImportReplicatorFailure() {
+    ReplicationSupervisor supervisor =
+        new ReplicationSupervisor(set, mutableReplicator,
+            newDirectExecutorService());
+
+    // Mock to fetch an exception in the importContainer method.
+    SimpleContainerDownloader moc =
+        Mockito.mock(SimpleContainerDownloader.class);
+    CompletableFuture<Path> res = new CompletableFuture<>();
+    res.complete(Paths.get("file:/tmp/file"));
```

Comment:
```suggestion
    res.complete(Paths.get("file:/tmp/no-such-file"));
```

This would document that the expected failure is due to `FileNotFoundException`.

Reply:
Changed